src/modules/admin/components/projectConfig/ProjectConfigDialog.tsx

```diff
@@ -0,0 +1,96 @@
+import { Box, DialogContent, DialogProps, DialogTitle } from '@mui/material';
+import React, { useMemo } from 'react';
+
+import CommonDialog from '@/components/elements/CommonDialog';
+import StaticForm from '@/modules/form/components/StaticForm';
+import {
+  CreateProjectConfigDocument,
+  CreateProjectConfigMutation,
+  CreateProjectConfigMutationVariables,
+  ProjectConfigInput,
+  StaticFormRole,
+  UpdateProjectConfigDocument,
+  UpdateProjectConfigMutation,
+  UpdateProjectConfigMutationVariables,
+  useGetProjectConfigsQuery,
+} from '@/types/gqlTypes';
+import { evictProjectConfigs } from '@/utils/cacheUtil';
+
+export interface ProjectDialogProps extends DialogProps {
+  projectId?: string;
+}
+
+const ProjectConfigDialog: React.FC<ProjectDialogProps> = ({
+  projectId,
+  ...props
+}) => {
+  const { data } = useGetProjectConfigsQuery();
+
+  const project = useMemo(
+    () => data?.projectConfigs?.nodes?.find((aec) => aec.id === projectId),
+    [data, projectId]
+  );
```

Comment:
This seems off to me - we already have the record from the table, so we do we need to query (even if it just hits the cache) to find it again.

Because we need these values in the dialog, I would switch to passing around the selected row (the Project Config reciord) rather than hyst the id. Eg:

ProjectConfigTable: `handleRowClick={onClickRow}`
ProjectConfigPage:
```
<ProjectConfigTable onRowClick={setSelected} />
{selected && <ProjectConfigDialog
        open={!!selected}
        config={selected}
        onClose={() => setSelected(null)}
      />}
```

Reply:
I see this was a pre-existing thing but I think its worth the change!

Reply:
oops response at https://github.com/greenriver/hmis-frontend/pull/665#discussion_r1508238149